topics/about_arrays.js

```diff
@@ -19,8 +19,10 @@ test("length", function() {
 test("splice", function() {
     var daysOfWeek = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
     var workingWeek = daysOfWeek.splice(__, __);
-    ok(workingWeek.equalTo([__]), 'what is the value of workingWeek?');
-    ok(daysOfWeek.equalTo([__]), 'what is the value of daysOfWeek?');
+    var weekend = daysOfWeek;
+
+    deepEqual(workingWeek, ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday'], 'what is the value of workingWeek?');
+    deepEqual(weekend, ['Saturday', 'Sunday'], 'what is the value of daysOfWeek?');
```

Comment:
There is because splice modifies the source table - let me explain in more details. 

In this case: `var workingWeek = daysOfWeek.splice(0, 5);` leaves `['Saturday', 'Sunday']` in `daysOfWeek` and returns days from 'Monday' to 'Friday' as a new array which we then assign to a variable named `workingWeek`. Since that name is misleading now I reassign reference kept in there to a variable named `weekend`. Just to make the `deepEqual` calls more welcoming ;)